src/Okta.Sdk/OktaClient.cs

```diff
@@ -78,6 +78,21 @@ public OktaClient(OktaClientConfiguration apiClientConfiguration = null, ILogger
             PayloadHandler.TryRegister<X509CaCertPayloadHandler>();
         }
 
+        private IOAuthTokenProvider GetTokenProvider(OktaClientConfiguration configuration, ILogger logger, ResourceFactory resourceFactory, IOAuthTokenProvider customTokenProvider)
+        {
+            switch (configuration.AuthorizationMode)
+            {
+                case AuthorizationMode.PrivateKey:
+                    return new DefaultOAuthTokenProvider(configuration, resourceFactory, logger: logger);
+                case AuthorizationMode.SSWS:
+                    return NullOAuthTokenProvider.Instance;
+                case AuthorizationMode.BearerToken:
+                    return new ExternalTokenProvider(configuration.BearerToken, customTokenProvider);
```

Comment:
shouldn't this be just customTokenProvider since it already implements the IOAuthTOkenProvider interface?

Reply:
customTokenProvider may be null here